### PR TITLE
PKCE vs Proxy Post

### PR DESCRIPTION
**Pull request by gcallaghan**
Doesn't quite compile and not sure why. I figured I'd send it sooner rather than later.

### CODE REVIEW

File: src/content/blog/pkce-vs-proxy.mdx [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+typing in email addresses to every site and updating it in multiple places if
+it changes. This is how “login with facebook” or “login with Google” buttons
+work.
+
+Unfortunately, there are a couple of inherent security risks with this setup.
+One service is not as trustworthy as another. You might want to allow that
+budgeting app to read your account balance, but you probably don’t want it to be
+able to empty your checking account. Additionally, the first time services
+communicate they must go over a public network where anybody can be listening.
+In order to protect credentials sent between services and what the service is
```
Comment (philsturgeon) [inline src/content/blog/pkce-vs-proxy.mdx]: the "and what the service is allowed to do with those credentials" part sounds a bit busted. 

Are we trying to say... in order to protect the credentials, and restrict usage of the service to... or something similar?

File: src/content/blog/pkce-vs-proxy.mdx [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+allowed to do with those credentials, some initial trust needs to be
+bootstrapped and protected ahead of time. This is done by combining two
+different components to ensure that the recipient of a temporary authorization
+code is, in fact, the one it is intended for. If you have heard of two-factor
+authentication(2FA), it is like that but for web services.
```
Comment (philsturgeon) [inline src/content/blog/pkce-vs-proxy.mdx]:
```suggestion
authentication (2FA), it is like that but for web services.
```

File: src/content/blog/pkce-vs-proxy.mdx [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+service then exchanges the authorization code combined with their client id and
+client secret for an access token and optionally a refresh token. The relying
+service then returns these to the user's client, usually a browser or mobile
+application. The client then includes this token to tell the resource server who
+the request is for and what permissions they have.
+
+### Proof of Key Code Exchange (PKCE)
+_Cryptographic Pixie Dust_
+
+With the rise and popularity of mobile and Single Page Applications (SPA) a new
```
Comment (philsturgeon) [inline src/content/blog/pkce-vs-proxy.mdx]:
```suggestion
With the rise and popularity of mobile apps, and Single Page Applications (SPA), a new
```

File: src/content/blog/pkce-vs-proxy.mdx [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+flow had to be invented for public clients. These clients cannot properly
+protect a client secret since they run in an environment not controlled by the
+application owner, either a mobile device or javascript in the browser. To
```
Comment (philsturgeon) [inline src/content/blog/pkce-vs-proxy.mdx]:
```suggestion
application owner, either a mobile device or JavaScript in the browser. To
```

File: src/content/blog/pkce-vs-proxy.mdx [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+mobile app the trust just isn’t there. The previous flows allow the for
+interaction outside of the initial client for exchanging the password with the
+Authorization server. For example, on a mobile device, the application might
+open a browser to begin the exchange and finish at an app-specific URL to
+complete it. This way the mobile application never has to handle the username
+and password directly.
+
+### What does this buy us?
+
+In both Authorization Code and PKCE flows 2 factors must be exchanged for valid
```
Comment (philsturgeon) [inline src/content/blog/pkce-vs-proxy.mdx]:
```suggestion
In both Authorization Code, and PKCE flows, two factors must be exchanged for valid
```

File: src/content/blog/pkce-vs-proxy.mdx [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+credentials. The authorization code, as presented as part of a redirect URL for
+consumption, along with some additional information posted in the body of the
+access token request. These 2 channels of communication mitigate a variety of
```
Comment (philsturgeon) [inline src/content/blog/pkce-vs-proxy.mdx]:
```suggestion
access token request. These two channels of communication mitigate a variety of
```

File: src/content/blog/pkce-vs-proxy.mdx [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+intercepted in flight.
+
+However, with PKCE there is one more consideration to take into account and that
+is where the Authorization server redirects after authentication. The client
+tells the authorization server where it wants to receive the code in the URL. If
+this is not an exact match, it means there is an opening for a malicious client
+to trick the Authorization Server to send the code somewhere else. That same
+client can also generate the temporary secret code.  While PKCE moves the
+goalposts in terms of difficulty, it still relies on this redirect as an anchor
+of trust. While having a fully qualified redirect URL is a best practice,
```
Comment (philsturgeon) [inline src/content/blog/pkce-vs-proxy.mdx]:
```suggestion
of trust. 

While having a fully qualified redirect URL is a best practice,
```

File: src/content/blog/pkce-vs-proxy.mdx (hunk 1 above)
Comment (gcallaghan) [reply src/content/blog/pkce-vs-proxy.mdx]: Yeah, its "protect the credentials and limit the capabilities of those credentials".

**Pull request merged**